proposals/XXXX-efficient-token-program.md

```diff
@@ -37,6 +37,29 @@ N/A.
 
 `p-token` is a like-for-like efficient re-implementation of the Token program. It is `no_std` (no heap memory allocations are made in the program) and uses zero-copy access for instruction and account data. Since it follows the same instructions and accounts layout, it does not require any changes to client code – it works as a drop-in replacement.
 
+Apart from the original SPL Token instructions, this proposal adds two additional instructions to the program:
+
+1. `withdraw_excess_lamports` (instruction discriminator `38`): allow recovering "bricked" SOL from mint accounts (e.g., USDC mint as `~323` SOL in excess). The logic of this instruction is the same as the current SPL Token-2022 instruction. 
+2. `batch` (instruction discriminator `255`): enable efficient CPI interaction with the Token program. This is a new instruction that can execute a variable number on Token instructions in a single invocation of the Token program. Therefore, the CPI invoke units (currently `1000` CU) are only consumed once, instead of each CPI instruction – this significantly improves the CUs required to perform multiple Token instructions in a CPI context.
+
+Note that `withdraw_excess_lamports` discriminator matches the value used in SPL Token-2022, while `batch` has a driscriminator value that is not used in either SPL Token nor Token-2022.
```

Comment:
Confirming that this is true! I went through all of the interfaces that token-2022 implements to be totally sure